logstash-core/api/lib/app.rb

```diff
@@ -1,19 +1,32 @@
 # encoding: utf-8
 require "cabin"
 require "logstash/json"
-require "app_helpers"
+require "helpers/app_helpers"
 require "app/service"
 require "app/command_factory"
+require "logstash/util/loggable"
+require "helpers/cabin_compatibility_layer"
 
 module LogStash::Api
   class BaseApp < ::Sinatra::Application
 
     attr_reader :factory
 
+    ::Cabin::Channel.class_eval do
+      include ::Cabin::Mixins::SinatraLogger
+    end
```

Comment:
Hi, I see this two methods as being very similar to what I did, isn't? what are the pros and cons of both for you? by using class_eval and include it make the definition very explicit, something less obscure than your option 1.

I do really see option 2 as doing the same as with the class eval, only thing is there is no module that we're actually including, just adding  a new method and per se not using the same mechanism as other cabin extensions.